mock-identity-system/src/main/java/io/mosip/esignet/mock/identitysystem/validator/IdentitySchemaValidator.java

```diff
@@ -0,0 +1,113 @@
+import javax.validation.ConstraintValidatorContext;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Set;
+
+@Component
+@Slf4j
+public class IdentitySchemaValidator implements ConstraintValidator<IdentitySchema, Object> {
+
+    @Value("${mosip.mock.ida.identity.schema.url}")
+    private String identitySchemaUrl;
+
+    @Value("#{${mosip.mock.ida.update-identity.non-mandatory.fields}}")
+    private Set<String> nonMandatoryFieldsOnUpdate;
+    private boolean isCreate;
+
+    private volatile JsonSchema schema;
+
+    @Autowired
+    ObjectMapper objectMapper;
+
+    @Autowired
+    ResourceLoader resourceLoader;
+
+
+    @Override
+    public void initialize(IdentitySchema constraintAnnotation) {
+        this.isCreate = constraintAnnotation.isCreate();
+    }
+
+    @Override
+    public boolean isValid(Object object, ConstraintValidatorContext context) {
+        context.disableDefaultConstraintViolation();
+        if (!(object instanceof RequestWrapper)) {
+            context.buildConstraintViolationWithTemplate("Invalid request type")
+                    .addConstraintViolation();
+            return false;
+        }
+        RequestWrapper wrapper= (RequestWrapper) object;
+        Object requestObject = wrapper.getRequest();
+        if (!(requestObject instanceof IdentityData)) {
+            context.disableDefaultConstraintViolation();
+            context.buildConstraintViolationWithTemplate("Invalid request object type").addConstraintViolation();
+            return false;
+        }
+        IdentityData identityData=(IdentityData) requestObject;
+        JsonNode identityJsonNode = objectMapper.valueToTree(identityData);
+        Set<ValidationMessage> errors = getSchema().validate(identityJsonNode);
+        boolean isValid=true;
+        String errorName="";
+        if(!isCreate){
+            for(ValidationMessage validationMessage: errors){
+                String field=validationMessage.
+                        getInstanceLocation().getName(0);
+                // Ignore validation errors with code 1029 (null value) for exempted fields when validating updateIdentity
+                if(!validationMessage.getCode().equals("1029") || !nonMandatoryFieldsOnUpdate.contains(field)){
+                    errorName="invalid_"+field.toLowerCase();
+                    isValid=false;
+                    break;
+                }
+            }
+        }else if(!errors.isEmpty()){
+            isValid=false;
+        }
+        if (!isValid) {
+            context.disableDefaultConstraintViolation();
+            if(StringUtils.isEmpty(errorName))
+             errorName="invalid_"+errors.iterator().next().getInstanceLocation().getName(0).toLowerCase();
+            context.buildConstraintViolationWithTemplate(errorName)
+                    .addConstraintViolation();
+            log.error("Validation failed for IdentityData: {}", errors);
+            return false;
```

Comment:
I think you can simplify this code. The logic for creating the errorCode from the errors list is duplicated.

Reply:
I was not able to figure out a way to simply the error code generation